File: Exo/ExoPHP/Rectangle/Cercle.php
<?php
class Cercle{

/********************Variable*******************/
private $_diametre;

/********************Accesseur*******************/
public function getDiametre()
{
return $this->_diametre;
}

public function setDiametre($diametre)
{
$this->_diametre = $diametre;
}

/********************Construct*******************/

public function __construct(array $options = []){
    if (!empty($options)) {
        $this->hydrate($options);
    }
}
public function hydrate($data){
    foreach ($data as $key => $value) {
        $methode = 'set' . ucfirst($key);
        if (is_callable([$this, $methode])) { 
            $this->$methode($value);
        }
    }
}

/********************Methode*******************/

    public function perimetre()  {
        return $this->_diametre*pi();
    }
    public function aire()  {
        return pi()*pow($this->_diametre/2,2);
    }
 
    public function afficher()  {
        echo $this; 
    }
    public function __toString(){
        $retour = "Diametre : ".$this->_diametre."cm - Périmètre : ".round($this->perimetre(),2)."cm - Aire : ".round($this->aire(),2)."cm²" ;
        return $retour."\n";
    }
}

File: Exo/exoCs/ExoClasse/Exo_Class_04/Cercle.cs
using System;
using System.Buffers.Text;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Exo_Class_04
{
    internal class Cercle
    {
        public int Diametre { get; set; }

        public Cercle(int diametre)
        {
            Diametre = diametre;
        }

        public override string ToString()
        {
            return "Diametre : " + Diametre + " - Périmètre : " + Math.Round(Perimetre(), 2) + " - Aire : " + Math.Round(Aire(), 2);
        }

        public Double Perimetre()
        {
            return double.Pi*Diametre;
        }
        public Double Aire()
        {
            return double.Pi*((Diametre/2) * (Diametre / 2));
        }


        public void AfficherCercle()
        {
            Console.WriteLine(this);
        }
    }
}
